src/Dialect/ONNX/ONNXOps/Tensor/Split.cpp

```diff
@@ -59,14 +59,25 @@ LogicalResult ONNXCommonSplitOpShapeHelper<OP_TYPE>::customComputeShape(
   } else {
     // If split parameter is not specified, the dimension is split to
     // equal-sized parts.
+    bool hasNumOutputsAttr = std::is_same_v<OP_TYPE, ONNXSplitOp>;
+    // TODO figure out how to handle when numResults is determined by
+    // num_outputs attribute introduced for Split in opset 18
+    // Currently the whole graph depends on the number of outputs being
+    // determined at the ONNX to ONNX-MLIR ingestion stage.
     IndexExpr splitInputDim = createIE->getShapeAsDim(input, axisIndex);
     LiteralIndexExpr numOfPartitions(numOfResults);
     if (splitInputDim.isLiteral() &&
-        (splitInputDim.getLiteral() % numOfResults != 0))
+        (splitInputDim.getLiteral() % numOfResults != 0) && !hasNumOutputsAttr)
```

Comment:
Yes thats a good point. It would be quite involved to do add this check to the verifier, because it would need to be added for SplitV11, and SplitV13. So we would need to create to new verify methods.

As a result I would need to change `gen_onnx_mlir.py` and regenerate a lot of code. I hope its ok that i leave this for a follow up PR.